test/src/unit-QueryCondition.cc

```diff
@@ -854,3 +854,123 @@ TEST_CASE(
 
   free(values);
 }
+
+TEST_CASE(
+    "QueryCondition: Test empty string", "[QueryCondition][empty_string]") {
+  const std::string field_name = "foo";
+  const uint64_t cells = 10;
+  const char* fill_value = "";
+  const Datatype type = Datatype::STRING_ASCII;
+  bool nullable = false;
+  bool var_size = true;
```

Comment:
It looks like most of the code is set up to handle `true` or `false`, so consider using `GENERATE` here to get coverage for both cases:
```
uint64_t cell_val_num = GENERATE(2, constants::var_num);
```
catch2 will generate tests for both paths  (x-ref [catch2 docs](https://github.com/catchorg/Catch2/blob/devel/docs/generators.md))

Reply:
Used GENERATE for var_size(true,false), nullable(true,false), and op(::EQ,::NE) to generate eight scenarios. thanks for the suggestion.

Reply:
👍